### PR TITLE
Gdr 2635

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Related JIRA issue: GDR-2635

## Why was it changed?
To be able to set unique identifiers

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

**Review by gladkia — APPROVED** [review]

File: R/standardize_MAE.R [hunk 1]
```diff
@@ -277,3 +277,123 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
...
+}
+
+#' Set Unique Drug Names
+#'
+#' This function sets the `DrugName`, `DrugName_2`, and `DrugName_3` fields in `rowData`
+#' to be unique by appending the corresponding `Gnumber`, `Gnumber_2`, and `Gnumber_3` in parentheses for duplicates.
+#'
+#' @param se A SummarizedExperiment object.
+#' @return A SummarizedExperiment object with unique `DrugName` fields in `rowData`.
+#' @examples
```
Comment (j-smola) [inline R/standardize_MAE.R:325]: Thanks for providing examples 👍

File: R/standardize_MAE.R [hunk 2]
```diff
@@ -277,3 +277,123 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
...
+#' se <- SummarizedExperiment::SummarizedExperiment(
+#'   assays = list(counts = matrix(1:9, ncol = 3)),
+#'   rowData = S4Vectors::DataFrame(DrugName = c("DrugA", "DrugA", "DrugB"),
+#'   Gnumber = c("G1", "G2", "G5"),
+#'   DrugName_2 = c("DrugC", "DrugC", "DrugD"),
+#'   Gnumber_2 = c("G3", "G4", "G5")
+#' ))
+#' se <- set_unique_drug_names(se)
+#' @export
```
Comment (j-smola) [inline R/standardize_MAE.R:334]: IMO This logic should be more sophisticated. Check:
```
 se <- SummarizedExperiment::SummarizedExperiment(
   assays = list(counts = matrix(1:9, ncol = 3)),
   rowData = S4Vectors::DataFrame(DrugName = c("DrugA", "DrugA", "DrugB"),
   Gnumber = c("G1", "G2", "G5"),
   DrugName_2 = c("DrugC", "DrugC", "DrugD"),
   Gnumber_2 = c("G3", "G3", "G5")
 ))
se <- set_unique_drug_names(se)
SummarizedExperiment::rowData(se)

DataFrame with 3 rows and 4 columns
     DrugName     Gnumber  DrugName_2   Gnumber_2
  <character> <character> <character> <character>
1  DrugA (G1)          G1  DrugC (G3)          G3
2  DrugA (G2)          G2  DrugC (G3)          G3
3       DrugB          G5       DrugD          G5
```
`DrugC` should stay as `DrugC` not as `DrugC (G3)`

File: R/standardize_MAE.R [hunk 3]
```diff
@@ -277,3 +277,123 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
...
+#' `colData` to be unique by appending the `clid` in parentheses for duplicates.
+#'
+#' @param se A SummarizedExperiment object.
+#' @return A SummarizedExperiment object with unique `CellLineName` in `colData`.
+#' @examples
+#' se <- SummarizedExperiment::SummarizedExperiment(
+#'   assays = list(counts = matrix(1:4, ncol = 2)),
+#'   colData = S4Vectors::DataFrame(CellLineName = c("ID1", "ID1"), clid = c("C1", "C2"))
+#' )
+#' se <- set_unique_cl_names(se)
```
Comment (j-smola) [inline R/standardize_MAE.R:294]: It is 100% sure that `colData` has only unique records? That such scenario is impossible?
```
> se <- SummarizedExperiment::SummarizedExperiment(
   assays = list(counts = matrix(1:4, ncol = 2)),
   colData = S4Vectors::DataFrame(CellLineName = c("ID1", "ID1"), clid = c("C1", "C1"))
 )

SummarizedExperiment::colData(se)
DataFrame with 2 rows and 2 columns
  CellLineName        clid
   <character> <character>
1          ID1          C1
2          ID1          C1

se <- set_unique_cl_names(se)
SummarizedExperiment::colData(se)

DataFrame with 2 rows and 2 columns
  CellLineName        clid
   <character> <character>
1     ID1 (C1)          C1
2     ID1 (C1)          C1
```

File: R/standardize_MAE.R [hunk 4]
```diff
@@ -277,3 +277,123 @@ refine_rowdata <- function(rd, se, default_v = "Undefined") {
...
+#' se <- set_unique_drug_names(se)
+#' @export
+#' @keywords standardize_MAE
+set_unique_drug_names <- function(se) {
+  checkmate::assertClass(se, "SummarizedExperiment")
+  
+  row_data <- SummarizedExperiment::rowData(se)
+  drug_columns <- unlist(get_env_identifiers(c("drug_name", "drug_name2", "drug_name3"), simplify = FALSE))
+  gnumber_columns <- unlist(get_env_identifiers(c("drug", "drug2", "drug3"), simplify = FALSE))
+
```
Comment (j-smola) [inline R/standardize_MAE.R:342]: Maybe to shorten loop:
```
row_data <- SummarizedExperiment::rowData(se)
drug_columns <- unlist(get_env_identifiers(c("drug_name", "drug_name2", "drug_name3"), simplify = FALSE))
drug_columns <-  intersect(drug_columns, names(row_data))
```

File: R/standardize_MAE.R (hunk 3 above)
Comment (bczech) [reply R/standardize_MAE.R:294]: it results from our data model

File: R/standardize_MAE.R (hunk 4 above)
Comment (j-smola) [reply R/standardize_MAE.R:342]: :chec

File: R/standardize_MAE.R (hunk 3 above)
Comment (j-smola) [reply R/standardize_MAE.R:294]: I had to ask. 😉 

✔️

**Pull request merged**